### PR TITLE
MGMT-17059 improved repository usability

### PR DESCRIPTION
**Pull request by celdrake**
https://issues.redhat.com/browse/MGMT-17059

Improvements added:

1. Creation of a new Repository:
- Form is now validated
- There's a checkbox for choosing public (no auth) or private (auth) repositories
- Added placeholder and helper text

2. Creation of a new Resource Sync:
- Form is now validated
- Added placeholder and helper text
 
3. Smaller improvements:
 - Repository: Changed "last transition date" to "X hours / minutes etc ago"
 - Repository details page: the repository is now a clickable link
 - Avoid using WarningIcon when not an error case, eg. password not set.
 - Added tooltip icon with explanation of what Resource syncs are (copy provisional)

https://github.com/flightctl/flightctl-ui/assets/829045/16293e6b-d80c-4cf7-8f25-ea3efe5b0377

### CODE REVIEW

File: src/app/components/Repository/RepositoryDetails.tsx [hunk 1]
```diff
@@ -55,14 +58,22 @@ const RepositoryDetails = () => {
...
                   <DescriptionListTerm>Url</DescriptionListTerm>
-                  <DescriptionListDescription>{repoDetails.spec.repo}</DescriptionListDescription>
+                  <DescriptionListDescription>
+                    <Button component="a" variant="link" isInline href={repoDetails.spec.repo} target="_blank">
+                      {repoDetails.spec.repo}
+                    </Button>
+
+                    <Icon status="info" style={{ marginLeft: '0.5rem' }}>
+                      <ExternalLinkAltIcon />
+                    </Icon>
```
Comment (rawagner) [inline src/app/components/Repository/RepositoryDetails.tsx:75]:
```suggestion
                    <Button component="a" variant="link" isInline href={repoDetails.spec.repo} target="_blank" icon={<ExternalLinkAltIcon />} iconPosition="end">
                      {repoDetails.spec.repo}
                    </Button>
```

File: src/app/components/Repository/CreateRepository/CreateRepository.tsx [hunk 2]
```diff
@@ -112,10 +155,12 @@ const CreateRepository = () => {
               name: '',
               url: '',
               credentials: {
-                user: '',
+                isPublic: true,
+                username: '',
                 password: '',
               },
             }}
+            validationSchema={repositorySchema}
```
Comment (rawagner) [inline src/app/components/Repository/CreateRepository/CreateRepository.tsx:163]: dont we want to disable submit btn when form is invalid ?

**Comment by jelkosz** [comment]
few comments

File: src/app/components/Repository/CreateRepository/CreateRepository.tsx (hunk 2 above)
Comment (celdrake) [reply src/app/components/Repository/CreateRepository/CreateRepository.tsx:163]: Good point, I will do the same also for the Resource sync creation.

File: src/app/components/Repository/CreateRepository/CreateRepository.tsx [hunk 3]
```diff
@@ -48,31 +76,46 @@ const getRepository = (values: RepositoryFormValues) => {
...
-          <TextInput aria-label="Name" value={values.name} onChange={(_, value) => setFieldValue('name', value)} />
+          <TextField name="name" aria-label="Name" value={values.name} />
         </FormGroup>
         <FormGroup label="Url" isRequired>
-          <TextInput aria-label="Url" value={values.url} onChange={(_, value) => setFieldValue('url', value)} />
+          <TextField
+            name="url"
+            aria-label="Url"
+            value={values.url}
+            placeholder="Example: git@github.com:flightctl/flightctl-demos.git"
```
[the remark this answers is not in the document]
Comment (celdrake) [reply src/app/components/Repository/CreateRepository/CreateRepository.tsx]: Done

**Review by rawagner — APPROVED** [review]

**Pull request merged**